src/Symfony/Component/Notifier/Channel/SmsChannel.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Channel;
+
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Notification\SmsNotificationInterface;
+use Symfony\Component\Notifier\Receiver\Receiver;
+use Symfony\Component\Notifier\Receiver\SmsReceiverInterface;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 5.0
+ */
+class SmsChannel implements ChannelInterface
+{
+    private $transport;
+
+    public function __construct(TransportInterface $transport)
+    {
+        $this->transport = $transport;
+    }
+
+    public function notify(Notification $notification, Receiver $receiver, string $transport = null, MessageBusInterface $bus = null): void
+    {
+        $message = null;
+        if ($notification instanceof SmsNotificationInterface) {
+            $message = $notification->asSmsMessage($receiver, $transport);
+        }
+
+        if (null === $message) {
+            $message = SmsMessage::fromNotification($notification, $receiver, $transport);
+        }
+
+        if (null !== $transport) {
+            $message->transport($transport);
+        }
+
+        if (null === $bus) {
+            $this->transport->send($message);
```

Comment:
There is not call to `$this->transport->supports($mesage)` ?

Reply:
No, not sure we should do it here. That would be a misconfiguration, and I wanted to avoid duplicating the error from the transport.